src/Models/ElementalArea.php

```diff
@@ -297,4 +297,42 @@ public function canView($member = null)
 
         return false;
     }
+
+    /**
+     * Return cacheability of the current elemental area instance
+     *
+     * This usually means every block in the current area is cacheable.
+     * If a single block is not cacheable, the whole area becomes not cacheable.
+     *
+     * @return bool
+     */
+    public function isCacheable()
+    {
+        foreach($this->Elements() as $element)
+        {
+            if (!$element->isCacheable()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns a unique cache key of the current elemental area instance
+     *
+     * @return string
+     */
+    public function getCacheKey()
+    {
+        $hash = hash_init('sha256');
+        hash_update($hash, $this->ID.$this->LastEdited);
```

Comment:
This cache key doesn't cover cases like `Versioned` and `Fluent` (where ID is not enough). I recommend using [getUniqueKey](https://github.com/silverstripe/silverstripe-framework/blob/4/src/ORM/DataObject.php#L4203) or similar.

Reply:
good catch, thank you!